docs/explanation/security.md

```diff
@@ -76,6 +76,7 @@ Otherwise, Ops doesn't introduce any new security risks. Ops does expand the imp
 * Use {external+juju:ref}`Juju secrets <secret>` for storing and sharing sensitive data.
 * Juju users that integrate a charm with a tracing receiver should also integrate with a certificate authority provider, to ensure all traces are sent via HTTPS.
 * Charms should follow best practices for writing secure Python code.
+* Charms should manage Unix permissions, including [`umask(2)`](https://manpages.ubuntu.com/manpages/noble/man2/umask.2.html) to respect the [principle of least authority](https://en.wikipedia.org/wiki/Principle_of_least_privilege).
```

Comment:
```suggestion
* Machine charms are responsible for setting appropriate ownership and permissions on the files and directories they create, for example using {py:func}`os.umask`.
```

Reasons for rewriting it so:
- certainly an issue for machine charms, not so much for k8s charms where pods are isolated, and some barriers exist between the containers; specifically everything in the charm container is run as the same user.
- "should" felt too prescriptive, I wanted to avoid the wall of "shoulds" and keep it more neutral/factual.